test/sqllogictest/array_fill.slt

```diff
@@ -0,0 +1,284 @@
+# by the Apache License, Version 2.0.
+
+# Misc. types
+
+query T
+SELECT array_fill(5, array[3])
+----
+{5,5,5}
+
+query T
+SELECT array_fill(5.1::double, array[3])
+----
+{5.1,5.1,5.1}
+
+query T
+SELECT array_fill(5.1, array[3])
+----
+{5.1,5.1,5.1}
+
+query T
+SELECT array_fill('z'::text, array[3])
+----
+{z,z,z}
+
+query T
+SELECT array_fill(INTERVAL '1d', array[3])
+----
+{"1 day","1 day","1 day"}
+
+query T
+SELECT array_fill('[1,)'::int4range, array[3])
+----
+{[1,),[1,),[1,)}
+
+query T
+SELECT (array_fill(5, array[3]))[1]
+----
+5
+
+# Lower bound adjustment works
+query T
+SELECT (array_fill(5, array[3], array[2]))[1]
+----
+NULL
+
+query T
+SELECT (array_fill(5, array[3], array[2]))[2]
+----
+5
+
+# Reveal structure of array
+query T
+SELECT
+    concat(
+        o,
+        ' ',
+        i,
+        ' ',
+        COALESCE(((array_fill(5, ARRAY[3, 2]))[o][i])::text, 'null')
+    )
+FROM generate_series(1, 4) AS o, generate_series(1, 4) AS i
+ORDER BY 1;
+----
+1 1 5
+1 2 5
+1 3 null
+1 4 null
+2 1 5
+2 2 5
+2 3 null
+2 4 null
+3 1 5
+3 2 5
+3 3 null
+3 4 null
+4 1 null
+4 2 null
+4 3 null
+4 4 null
+
+query T
+SELECT array_fill(5, array[3, 2], array[2, 3])
+----
+{{5,5},{5,5},{5,5}}
+
+# Reveal structure of 2D array
+query T
+SELECT
+    concat(
+        o,
+        ' ',
+        i,
+        ' ',
+        COALESCE(((array_fill(5, ARRAY[3, 2], ARRAY[2, 3]))[o][i])::text, 'null')
+    )
+FROM generate_series(1, 4) AS o, generate_series(1, 4) AS i
+ORDER BY 1;
+----
+1 1 null
+1 2 null
+1 3 null
+1 4 null
+2 1 null
+2 2 null
+2 3 5
+2 4 5
+3 1 null
+3 2 null
+3 3 5
+3 4 5
+4 1 null
+4 2 null
+4 3 5
+4 4 5
+
+# Reveal structure of 3D array
+query T
+SELECT
+    concat(
+        a,
+        ' ',
+        b,
+        ' ',
+        c,
+        ' ',
+        COALESCE(((array_fill(5, ARRAY[3, 2, 1], ARRAY[1, 2, 3]))[a][b][c])::text, 'null')
+    )
+FROM
+    generate_series(1, 3) AS a,
+    generate_series(1, 3) AS b,
+    generate_series(1, 3) AS c
+ORDER BY 1;
+----
+1 1 1 null
+1 1 2 null
+1 1 3 null
+1 2 1 null
+1 2 2 null
+1 2 3 5
+1 3 1 null
+1 3 2 null
+1 3 3 5
+2 1 1 null
+2 1 2 null
+2 1 3 null
+2 2 1 null
+2 2 2 null
+2 2 3 5
+2 3 1 null
+2 3 2 null
+2 3 3 5
+3 1 1 null
+3 1 2 null
+3 1 3 null
+3 2 1 null
+3 2 2 null
+3 2 3 5
+3 3 1 null
+3 3 2 null
+3 3 3 5
+
+# Polymorphic solution
+
+query error db error: ERROR: could not determine polymorphic type because input has type unknown
+SELECT array_fill(null, array[3])
+
+query error db error: ERROR: could not determine polymorphic type because input has type unknown
+SELECT array_fill(null, null)
+
+query T
+SELECT array_fill(null::int, array[3])
+----
+{NULL,NULL,NULL}
+
+# Prohibited types
+
+query error db error: ERROR: array_fill with arrays not yet supported
+SELECT array_fill(ARRAY[1], array[3, 2])
+
+query error db error: ERROR: array_fill with arrays not yet supported
+SELECT array_fill(ARRAY[1], array[3, 2], array[2, 3])
```

Comment:
PostgreSQL also doesn't support this, mercifully.

Reply:
Small blessings, though the error it returns is odd.